discovery/manager.go

```diff
@@ -146,7 +181,9 @@ func (m *Manager) runProvider(ctx context.Context, poolKey poolKey, updates chan
 			if !ok {
 				return
 			}
-			m.updateGroup(poolKey, tgs)
+			for _, s := range p.subs {
+				m.updateGroup(poolKey{setName: s, provider: p.name}, tgs)
```

Comment:
why not use the `p *provider` as a pool key directly?  I used a seperate struct just to make it unique enough, but now with the `type provider struct` maybe we can use it directly and remove the `poolKey type` entirely.

Reply:
Hmm we still need to dispatch the target groups from 1 provider to N jobs at some point.